### PR TITLE
Bugfix/3673 644 info list item console error

### LINKED ISSUE #644: InfoListItem should disable typography wrappers
#### Describe the bug
InfoListItem throws invalid DOM nesting errors because we are using custom components inside the ListItemText

#### What is the expected behavior?
There should be no DOM Validation warnings

#### What are the steps to reproduce?
1. Pass custom elements to the title/subtitle/info props for InfoListItem
2. Observe the errors in the console

#### Screenshots / Screen recording
image

#### Code snippet / Link to minimum reproduction example

```tsx
<InfoListItem
    title={
        <Box>
            <Typography noWrap>
                {title}
            </Typography>
        </Box>
    }
    subtitle={description}
/>
```

#### Your environment information

N/A

#### Suggested fix

https://github.com/brightlayer-ui/react-component-library/blob/dev/components/src/core/InfoListItem/InfoListItem.tsx#L254
Add the `disableTypography` prop to the ListItemText

### PR DESCRIPTION
**Pull request by bkarambe**
Fixes #644  BLUI-3673

#### Changes proposed in this Pull Request:
- Added component div to title, subtitle and info typography

#### Screenshots:
Screenshot 2022-12-26 at 5 27 37 PM

#### To Test:
- Checkout [bugfix/info-list-item-console-error-example](https://github.com/brightlayer-ui/react-showcase-demo/tree/bugfix/info-list-item-console-error-example) branch in react showcase demo
- build the components
- paste the dist folder contents to showcase demo's node_modules/@brighlayer-ui/react-components/
- cd demos/showcase
- yarn start

**Note: Adding disableTypography props will not allow to pass primaryTypographyProps or secondaryTypographyProps, which will affect the styles of title, subtitle and info**

### CODE REVIEW

File: components/src/core/InfoListItem/InfoListItemStyledComponents.tsx [hunk 1]
```diff
@@ -166,11 +166,13 @@ export const Info = styled(Typography, {
     name: 'info-list-item',
     slot: 'info',
     shouldForwardProp: (prop) => prop !== 'fontColor',
-})<Pick<InfoListItemProps & TypographyProps, 'fontColor' | 'noWrap'>>(({ fontColor, theme }) => ({
-    fontWeight: 400,
-    lineHeight: 1.3,
-    color: fontColor || (theme.palette.mode === 'dark' ? theme.palette.text.secondary : 'inherit'),
-}));
+})<Pick<InfoListItemProps & TypographyProps & BoxProps, 'fontColor' | 'component' | 'noWrap'>>(
```
Comment (joebochill) [inline components/src/core/InfoListItem/InfoListItemStyledComponents.tsx:169]: Why do you need BoxProps in here?

**Comment by joebochill** [comment]
> **Note: Adding disableTypography props will not allow to pass primaryTypographyProps or secondaryTypographyProps, which will affect the styles of title, subtitle and info**

Does this mean switching the component to 'div' is sufficient to fix the dom nesting warnings, or does this mean that this is an incomplete solution?

**Comment by bkarambe** [comment]
> > **Note: Adding disableTypography props will not allow to pass primaryTypographyProps or secondaryTypographyProps, which will affect the styles of title, subtitle and info**
> 
> Does this mean switching the component to 'div' is sufficient to fix the dom nesting warnings, or does this mean that this is an incomplete solution?

Yes, switching to 'div' only is sufficient to fix dom nesting warnings.

**Comment by huayunh** [comment]
changelog? version number?

**Comment by huayunh** [comment]
Somehow I was unable to trigger a package publishing after changing the package version in package.json to `6.1.3-alpha.0` (prev published version was `6.1.2`). One of you guys might be able to figure this out better than me...

**Comment by huayunh** [comment]
It looks like this does not fix the problem in the bug report at all. The bug report talks about the `title` prop in the Info List Item, yet your fix is targeting `subtitle` and `info` prop.

I believe the original bug was logged while Joe was doing MUIv5 updates for docit. To see the bug, plug the latest `dist` into docit's `node_modules`, run it locally, go to the developer resources page, and check the console log.

File: components/src/core/InfoListItem/InfoListItemStyledComponents.tsx (hunk 1 above)
Comment (huayunh) [reply components/src/core/InfoListItem/InfoListItemStyledComponents.tsx:169]: I don't understand... why does `info` have to do with `BoxProps`? And would you please elaborate what you mean by "we were not passing before"?

**Comment by bkarambe** [comment]
> It looks like this does not fix the problem in the bug report at all. The bug report talks about the `title` prop in the Info List Item, yet your fix is targeting `subtitle` and `info` prop.
> 
> I believe the original bug was logged while Joe was doing MUIv5 updates for docit. To see the bug, plug the latest `dist` into docit's `node_modules`, run it locally, go to the developer resources page, and check the console log.

I checked out bugfix/3673-644-info-list-item-console-error  ran yarn build and pasted the react-components in doc-it and navigates to developer resources page, but I did not see any error related to DOM nesting errors.

**Comment by bkarambe** [comment]
@huayunh The main issue with InfoListItem was it is using Typography for title, subtitle and info props, which by default uses **p tag**. Using **Box or Typography inside Typography** throws console warnings. I added a fix passing **component=div** to typography. We were not passing component to Info before, that's why we had to add component from BoxProps.

**Comment by huayunh** [comment]
>  but I did not see any error related to DOM nesting errors.

My apologies. I retried using a fresh clone which seem to make the issue go away.

**Review by huayunh — APPROVED** [review]

**Pull request merged**